### PR TITLE
fix: Cannot use an API Key to add users to a group

### LINKED ISSUE #4361: AttributeError: 'APIKeyUser' object has no attribute 'is_group_admin'
Sentry Issue: [FLAGSMITH-API-4S5](https://flagsmith.sentry.io/issues/5622561504/?referrer=github_integration)

```
AttributeError: 'APIKeyUser' object has no attribute 'is_group_admin'
(7 additional frame(s) were not displayed)
...
  File "organisations/permissions/permissions.py", line 154, in has_object_permission
    if request.user.is_group_admin(obj.id):
```

### PR DESCRIPTION
**Pull request by matthewelwell**
## Changes

Fixes #4361 

## How did you test this code?

Added a new test for the specific failure case and also updated the test module to use admin_client_new in all tests and resolved any other similar issues.

### CODE REVIEW

File: api/api_keys/user.py [hunk 1]
```diff
@@ -64,6 +64,11 @@ def is_project_admin(self, project: "Project") -> bool:
     def is_environment_admin(self, environment: "Environment") -> bool:
         return is_master_api_key_environment_admin(self.key, environment)
 
+    def is_group_admin(self, group_id: int) -> bool:
+        # This is added to match the interface of the FFAdminUser model,
+        # but a MasterAPIKey cannot be a group admin.
+        return False
```
Comment (khvn26) [inline api/api_keys/user.py:70]: Can we add this to `UserABC`?

**Comment by khvn26** [comment]
Can we accommodate `edge_api.identities.tasks.call_environment_webhook_for_feature_state_change` and fix #4370 as part of this PR as well?

**Comment by matthewelwell** [comment]
> Can we accommodate `edge_api.identities.tasks.call_environment_webhook_for_feature_state_change` and fix #4370 as part of this PR as well?

Well, strictly, that would involve changing the branch name and recreating the PR, etc. - I can do it, but at that point, we might as well just create a new PR?

**Comment by khvn26** [comment]
> Well, strictly, that would involve changing the branch name and recreating the PR, etc

Or just changing the PR description to "Fixes #4361, #4370"?

**Comment by matthewelwell** [comment]
> > Well, strictly, that would involve changing the branch name and recreating the PR, etc
> 
> Or just changing the PR description to "Fixes #4361, #4370"?

Hence 'strictly'. It's ugly to _not_ change the branch name, but I can. I guess the branch name doesn't live anywhere post merge so it's fine. I'll just do it, yep.

**Comment by matthewelwell** [comment]
Actually @khvn26, sorry, but no, #4370 is actually quite a bit more involved as it breaks the webhook schema. If you look [here](https://github.com/Flagsmith/flagsmith/blob/main/api/edge_api/identities/tasks.py#L46), we're currently _always_ sending the email of the user that made the change. We will no longer have that option available to us. 

Based on this, I'd say it definitely deserves its own PR.

**Review by khvn26 — APPROVED** [review]

**Pull request merged**